model/products.model.js

```diff
@@ -1,17 +1,29 @@
 const mongoose = require("mongoose");
 const { Schema } = mongoose;
+require("mongoose-type-url");
 
-const productSchema = new Schema({
-	name: String,
-	price: Number,
-	image: String,
-	description: String,
-	ratings: String,
-	delivery: String,
-	offer: String,
-	inStock: { type: Boolean },
-	cartListed: { type: Boolean, default: false },
-	wishListed: { type: Boolean, default: false },
-});
+const productSchema = new Schema(
+	{
+		brand: { type: String, required: true },
+		name: { type: String, required: true },
+		modelNo: { type: String, required: true, unique: true },
+		price: { type: Number, required: true },
+		imageUrl: { type: mongoose.SchemaTypes.Url, required: true },
+		description: {
+			type: String,
+			minLength: [100, "Min length should be 100 characters"],
+			required: true,
+		},
+		ratings: { type: String, required: true },
+		delivery: { type: String, required: true },
+		offer: { type: String, required: true },
+		inStock: { type: Boolean, required: true },
+		cartListed: { type: Boolean, default: false },
+		wishListed: { type: Boolean, default: false },
```

Comment:
Having a seperate cart and wishlist model would better I feel .Because products will be same for every user.

Reply:
done